test/e2e/fail_forward_e2e_test.go

```diff
@@ -102,14 +199,39 @@ var _ = Describe("Fail Forward Upgrades", func() {
 			_, err = fetchCSV(crclient, subscription.Status.CurrentCSV, generatedNamespace.GetName(), buildCSVConditionChecker(operatorsv1alpha1.CSVPhaseSucceeded))
 			Expect(err).ShouldNot(HaveOccurred())
 
-			By("updating the operator group to use the service account without required permissions to simulate InstallPlan failure")
-			Eventually(operatorGroupServiceAccountNameSetter(crclient, generatedNamespace.GetName(), ogName, saNameWithNoPerms)).Should(Succeed())
-
-			By("updating the catalog with v0.2.0 bundle image")
-			brokenProvider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, failForwardTestDataBaseDir, "example-operator.v0.2.0.yaml"))
-			Expect(err).To(BeNil())
-			err = magicCatalog.UpdateCatalog(context.Background(), brokenProvider)
+			By("removing the existing catalog source")
+			err = ctx.Ctx().OperatorClient().OperatorsV1alpha1().CatalogSources(generatedNamespace.GetName()).Delete(context.Background(), catalogSource.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("removing the previous catalog source pod(s)")
+			Eventually(func() (bool, error) {
+				listOpts := metav1.ListOptions{
+					LabelSelector: "olm.catalogSource=" + catalogSource.GetName(),
+					FieldSelector: "status.phase=Running",
+				}
+				fetched, err := ctx.Ctx().KubeClient().KubernetesInterface().CoreV1().Pods(catalogSource.GetNamespace()).List(context.Background(), listOpts)
+				if err != nil {
+					return false, err
+				}
+				if len(fetched.Items) == 0 {
+					return true, nil
+				}
+				ctx.Ctx().Logf("waiting for the catalog source %s pod to be deleted...", fetched.Items[0].GetName())
+				return false, nil
+			}).Should(BeTrue())
+
+			By("updating the catalog with a v0.2.0 bundle image that has an invalid CRD")
+			pms, crds, csvs, err = loadData("v0.1.0", "v0.2.0-invalid-crd")
```

Comment:
Behavior change? Removing the prior catalog rather than replacing, and using `v0.2.0-invalid-crd` rather than `v0.2.0`?

Reply:
> Removing the prior catalog rather than replacing

If you read the implementation of the magic catalog, it's identical: 

https://github.com/operator-framework/operator-lifecycle-manager/blob/1ca6d9e799cd7da2bc8da287b310da76846efeb2/test/e2e/magic_catalog.go#L84-L91

> using v0.2.0-invalid-crd rather than v0.2.0?

That is the point of the PR, yes. We trigger a failing InstallPlan by having invalid data in the bundle as opposed to missing permissions.